nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/util/SFTPTransfer.java

```diff
@@ -317,6 +317,12 @@ protected void getListing(final String path, final int depth, final int maxResul
             final RemoteResourceFilter filter = (entry) -> {
                 final String entryFilename = entry.getName();
 
+                // Since SSHJ does not have the concept of BREAK that JSCH had, we need to move this before the call to listing.add
+                // below, otherwise we would keep adding to the listings since returning false here doesn't break
+                if (listing.size() >= maxResults) {
+                    return false;
+                }
+
```

Comment:
Is this change necessary? It seems useful in some circumstances, but not needed for this particular property change.